app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def self.update_posts_counter(name)
+    user = User.find_by(name:)
+    user.update(posts_counter: Post.where(author_id: User.find_by(name:).id).length)
+  end
```

Comment:
- [ ] [optional] @sambeck87 You could also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private , this could also apply to other counters.
```suggestion
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```